libbeat/monitoring/inputmon/httphandler.go

```diff
@@ -36,19 +38,31 @@ const (
 )
 
 type handler struct {
-	registry *monitoring.Registry
+	registryDataset        *monitoring.Registry
+	registryInternalInputs *monitoring.Registry
 }
 
 // AttachHandler attaches an HTTP handler to the given mux.Router to handle
 // requests to /inputs.
-func AttachHandler(r *mux.Router) error {
-	return attachHandler(r, globalRegistry())
+func AttachHandler(beatInfo beat.Info, r *mux.Router) error {
+	intInputsReg := beatInfo.Monitoring.Namespace.GetRegistry().
```

Comment:
Are you sure that neither `beatInfo.Monitoring.Namespace` nor `beatInfo.Monitoring` will ever be `nil`?

IIRC I've seen assumptions like that breaking Beats, it was wither a test or one of our "debug input/output", so not really  a "production issue", but it would be nice to make sure we're well protected here.